src/expression/definitions/interpolate.ts

```diff
@@ -171,9 +174,15 @@ class Interpolate implements Expression {
         const outputLower = outputs[index].evaluate(ctx);
         const outputUpper = outputs[index + 1].evaluate(ctx);
 
+        if (this.type.kind === 'projection') {
```

Comment:
Is this if needed given the below switch case? I find this "if" odd here...

Reply:
Alternatively, we can do :

```ts
 switch (this.operator) {
            case 'interpolate':
 //  The line below this complains if typescript thinks this.type.kind can equal "projection"
                return interpolate[this.type.kind](outputLower, outputUpper, t);  
            case 'interpolate-hcl':
                return interpolate.color(outputLower, outputUpper, t, 'hcl');
            case 'interpolate-lab':
                return interpolate.color(outputLower, outputUpper, t, 'lab');
        }
```